### PR TITLE
Implement cupy.argsort

### PR DESCRIPTION
**Pull request by takagi**
This PR implements `cupy.argsort` following the implementation of `cupy.sort` in #1817 .

### CODE REVIEW

File: cupy/cuda/cupy_thrust.cu [hunk 1]
```diff
@@ -22,3 +28,39 @@ template void cupy::thrust::sort<cpy_long>(void *, ssize_t);
...
+ * argsort
+ */
+
+template <typename T>
+class elem_less {
+public:
+    elem_less(void *data):_data((T *)data) {}
+    __device__ bool operator()(ssize_t i, ssize_t j) { return _data[i] < _data[j]; }
+private:
+    T *_data;
```
Comment (unnonouno) [inline cupy/cuda/cupy_thrust.cu]: `const T` ?

Comment (takagi) [inline cupy/cuda/cupy_thrust.cu]: Yes, it should be. I have added `const` qualifier.

**Comment by alexbw** [comment]
Bump, this would be useful in a project of mine

**Comment by MannyKayy** [comment]
Another bump, very useful to me. Needed for a current project

**Comment by takagi** [comment]
This PR reposted to https://github.com/cupy/cupy/pull/67 after CuPy separation from Chainer, so I close this.

**Pull request closed without merging**